pkg/codegen/internal/test/testdata/output-funcs/dotnet/funcWithDefaultValue.cs

```diff
@@ -0,0 +1,75 @@
+// *** WARNING: this file was generated by . ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Pulumi.Serialization;
+
+namespace Pulumi.MadeupPackage.Codegentest
+{
+    public static class FuncWithDefaultValue
+    {
+        /// <summary>
+        /// Check codegen of functions with default values.
+        /// </summary>
+        public static Task<FuncWithDefaultValueResult> InvokeAsync(FuncWithDefaultValueArgs args, InvokeOptions? options = null)
+            => Pulumi.Deployment.Instance.InvokeAsync<FuncWithDefaultValueResult>("madeup-package:codegentest:funcWithDefaultValue", args ?? new FuncWithDefaultValueArgs(), options.WithVersion());
+
+        public static Output<FuncWithDefaultValueResult> Invoke(FuncWithDefaultValueOutputArgs args, InvokeOptions? options = null)
+        {
+            return Pulumi.Output.All(
+                args.A.Box(),
+                args.B.Box()
+            ).Apply(a => {
+                    var args = new FuncWithDefaultValueArgs();
+                    a[0].Set(args, nameof(args.A));
+                    a[1].Set(args, nameof(args.B));
+                    return InvokeAsync(args, options);
+            });
+        }
+    }
+
+
+    public sealed class FuncWithDefaultValueArgs : Pulumi.InvokeArgs
+    {
+        [Input("a", required: true)]
+        public string A { get; set; } = null!;
+
+        [Input("b")]
+        public string? B { get; set; }
+
+        public FuncWithDefaultValueArgs()
+        {
+            B = "b-default";
+        }
+    }
+
+    public sealed class FuncWithDefaultValueOutputArgs
+    {
+        [Input("a", required: true)]
```

Comment:
These attributes aren't used on the `XxxOutputArgs` variants, right? Should we not emit them?

Reply:
My initial instinct (now added to the PR) is to drop them as they're not enforced, but alternatively we can use this to try to enforce required values via reflection. I think that's what inheriting from InvokeArgs used to do, but I got rid of it because as written it was not letting inputs pass through. Let me see..

Reply:
Never mind. The required arguments are enforced by dynamic checks already on the underlying function. So I think skipping the Input is ok.